### PR TITLE
shim: add basic ci test

### PR DESCRIPTION
**Pull request by haosanzi**

### CODE REVIEW

File: .github/workflows/basic.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  # shim basic test
+  shim:
+    name: shim basic test
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        go-version: [1.16.x, 1.17.x]
```
Comment (jodh-intel) [inline .github/workflows/basic.yaml:21]: ooi, why no 1.18.x and 1.19.x?

**Review by jodh-intel — APPROVED** [review]
Thanks @haosanzi.

File: .github/workflows/basic.yaml (hunk 1 above)
Comment (mythi) [reply .github/workflows/basic.yaml:21]: +1

File: .github/workflows/basic.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Run unit test
+        run: |
+          go test -v ./...
+        working-directory: src/shim
+
+      - name: Run go fmt check
+        uses: Jerome1337/gofmt-action@v1.0.4
+        with:
+          gofmt-path: 'src/shim'
+          gofmt-flags: '-l -d'
```
Comment (mythi) [inline .github/workflows/basic.yaml:44]: My preference would be `golangci-lint` with the appropriate config:
```suggestion
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3
        with:
          version: v1.48.0
          args: -v --timeout 5m
          working-directory: 'src/shim'
```

**Review by mythi — APPROVED** [review]
LGTM, thanks!

**Review by dcmiddle — APPROVED** [review]

**Pull request merged**